### PR TITLE
feat: added tree file store implementation.

### LINKED ISSUE #31: Enhancement:  organise filestore packages in sub directory structure
As the amount of uploads increases, it would be nicer to store uploads in a hierarchical sub directory structure to make its easier to navigate and backup for administrators. Equally theoretically some older file systems have file/folder limits that in extreme cases could be exceeded if this project ever evolved into caching/proxying upstream downloads also.

One such structure inspired by the current api could be something like:

`$baseDir/packages/$package/versions/$package-$version.tar.gz`

It may be worth adding this as a new implementation of PackageStore class so that users of the api have the flexibility to decide between using the existing way to store uploads, or use the new way.

### PR DESCRIPTION
**Pull request by dellgreen**
closes #31

### CODE REVIEW

**Comment by pd4d10** [comment]
Thanks for the contribution!

There seems to be similar with the `FileStore`. How about adding a parameter?

**Comment by dellgreen** [comment]
How about like this?

File: unpub/lib/src/file_store.dart [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
 import 'package:path/path.dart' as path;
+
 import 'package_store.dart';
 
 class FileStore extends PackageStore {
   String baseDir;
+  bool asTree;
 
-  FileStore(this.baseDir);
+  FileStore(this.baseDir, {this.asTree: false});
```
Comment (pd4d10) [inline unpub/lib/src/file_store.dart:11]: Would it be more general to replace this param with `filePath`?

File: unpub/lib/unpub.dart [hunk 2]
```diff
@@ -3,4 +3,4 @@ export 'src/mongo_store.dart';
 export 'src/package_store.dart';
 export 'src/file_store.dart';
 export 'src/app.dart';
-export 'src/models.dart';
+export 'src/models.dart';
```
Comment (pd4d10) [inline unpub/lib/unpub.dart:6]: We'd better keep the new line at the file end

File: unpub/lib/src/file_store.dart (hunk 1 above)
Comment (pd4d10) [reply unpub/lib/src/file_store.dart:11]: For example,

```dart
String Function(String name, String version)? getFilePath
```

Comment (dellgreen) [reply unpub/lib/src/file_store.dart:11]: Sorry, not sure i follow. Given that its a dynamically generated path based upon the current package, I don't see how its user configurable?

Comment (pd4d10) [reply unpub/lib/src/file_store.dart:11]: Yeah, it depends on the package name and version. We can pass a function to do it.

Comment (dellgreen) [reply unpub/lib/src/file_store.dart]: ah, i think is see now what you are suggesting

**Comment by dellgreen** [comment]
i think i need to change it slightly so that the getFilePath function is concatenated to basedir as basedir is also being passed to the constructor otherwise its a bit confusing to the user.

**Pull request merged**